aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -21,14 +21,15 @@ __device__ inline int min(int a, int b) {
 __device__ inline int max(int a, int b) {
   return a >= b ? a : b;
 }
-
-template <typename scalar_t, typename accscalar_t, bool COUNT_INCLUDE_PAD, bool USE_DIVISOR>
+  
+template <typename scalar_t, typename accscalar_t>
```

Comment:
Perf regression from this seems to be all in cases where both B C are tiny. Now that we have the switch statement on kernel calls, the complicated template parameter does make the code quite cumbersome.

I'll let @ngimel or @VitalyFedyunin to make the call. Here's the linke to Xiao's comment with benchmarking https://github.com/pytorch/pytorch/pull/35855#discussion_r405102773